drivers/serial/uart_stm32.c

```diff
@@ -1957,6 +1957,33 @@ static int uart_stm32_registers_configure(const struct device *dev)
 	}
 #endif
 
+#ifdef USART_CR1_FIFOEN
+	if (config->fifo_enable) {
+		LL_USART_EnableFIFO(config->usart);
```

Comment:
Maybe it is worth setting `LL_USART_SetTXFIFOThreshold` and ` LL_USART_SetRXFIFOThreshold` explicitly or via DT?